fxa-client/src/lib.rs

```diff
@@ -653,7 +653,14 @@ mod tests {
         let url = fxa.begin_pairing_flow(&PAIRING_URL, &["https://identity.mozilla.com/apps/oldsync"]);
 
         assert!(url.is_err());
-        assert_eq!(format!("{:?}", url), "Err(Error(\n\nOrigin mismatch))")
+
+        match url {
+            Ok(url) => { assert!(false, "should have error"); }
+            Err(err) => match err.kind() {
+                ErrorKind::OriginMismatch { .. } => assert!(true),
```

Comment:
You also probably should just delete the `assert!(true)` and replace it with `{}` or something, `assert!(true)` is a no-op.

I don't really care about this one though.